wagtail/wagtailforms/templates/wagtailforms/index_submissions.html

```diff
@@ -23,11 +23,72 @@
                 },
                 lang: 'lang'
             });
+
+            var selectAllCheckbox = document.getElementById('select-all');
+            var deleteButton = document.getElementById('delete-submissions');
+            var selectedSubmissions = {};
```

Comment:
Yeah, this was a little bit over-engineered. I've simplified it as per your comment.

I couldn't reproduce a situation where the checkboxes would be out of sync when the user presses the back button. For me, the checkboxes are always reset to their unchecked state.